### ISSUE TITLE
`yaml.ConfigGroup` raises exception when run with Pulumi 3.4.0 under Python 

### ISSUE THREAD

**Issue by csssuf** [pulumi/pulumi-kubernetes#1602]
When running with Pulumi 3.4.0 on Python, trying to create a `yaml.ConfigGroup` fails with an exception in the ConfigGroup constructor. Downgrading to Pulumi 3.3.1 fixes the issue, which suggests pulumi/pulumi#7175 may be related.

## Steps to reproduce

Create a `ConfigGroup`:

1. `__main__.py`:
```python
"""A Kubernetes Python Pulumi program"""

import pulumi
from pulumi_kubernetes.yaml import ConfigGroup

CONFIGS = [
    """
apiVersion: v1
kind: Namespace
metadata:
  name: java
    """.strip(),
    """
apiVersion: v1
kind: Namespace
metadata:
  name: chesterfield-kings
    """.strip(),
]

config_group = ConfigGroup("configs", yaml=CONFIGS)
```
2. `pulumi preview`

Expected: The preview runs successfully and previews creating the two namespaces.
Actual:
```
error: Program failed with an unhandled exception:
error: Traceback (most recent call last):
  File "/pulumi/bin/pulumi-language-python-exec", line 92, in <module>
    loop.run_until_complete(coro)
...
[long RPC stacktrace snipped]
...
  File "/usr/local/lib/python3.8/site-packages/pulumi/output.py", line 157, in run
    value = await self._future
  File "/usr/local/lib/python3.8/site-packages/pulumi/output.py", line 178, in run
    transformed: Input[U] = func(value)
  File "/usr/local/lib/python3.8/site-packages/pulumi_kubernetes/yaml.py", line 190, in <lambda>
    self.resources = pulumi.Output.all(resources, self.resources).apply(lambda x: {**x[0], **x[1]})
TypeError: 'list' object is not a mapping
```

**Comment by jeid64** [comment]
@lukehoban your PR may have broken ConfigGroup in this provider

**Comment by lblackstone** [comment]
I just noticed that this is also causing k8s CI tests to fail: https://github.com/pulumi/pulumi-kubernetes/pull/1603/checks?check_run_id=2777986718#step:26:2492

It looks like we're not running nightly tests for some reason, so I'm not sure exactly when this regressed.

**Comment by lblackstone** [comment]
@lukehoban Do you know if this will be a quick fix, or should I pin the pulumi dependency to `v3.3.1` in the Python SDK for now?

**Comment by lukehoban** [comment]
The Kubernetes issues is fixed in 3.3.1, and the underlying issue is fixed in https://github.com/pulumi/pulumi/issues/7252.
